cmd/mdatagen/validate_test.go

```diff
@@ -101,3 +105,48 @@ func TestValidate(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateMetricDuplicates(t *testing.T) {
+	allowedMetrics := map[string][]string{
+		"container.cpu.utilization": {"docker_stats", "kubeletstats"},
+		"container.memory.rss":      {"docker_stats", "kubeletstats"},
+		"container.uptime":          {"docker_stats", "kubeletstats"},
+	}
+	allMetrics := map[string][]string{}
+	err := filepath.Walk("../../receiver", func(path string, info fs.FileInfo, err error) error {
+		if info.Name() == "metadata.yaml" {
+			md, err := loadMetadata(path)
+			assert.NoError(t, err)
+			if len(md.Metrics) > 0 {
+				for metricName := range md.Metrics {
+					allMetrics[md.Type] = append(allMetrics[md.Type], string(metricName))
+				}
+			}
+		}
+		return nil
+	})
+	assert.NoError(t, err)
+
+	seen := make(map[string]string)
+	for receiver, metrics := range allMetrics {
+		for _, metricName := range metrics {
+			val, exists := seen[metricName]
```

Comment:
This looks a bit difficult to follow, I'd suggest
```
if val, exists := seen[metricName]; exists {
  // do the validation here
}
seen[metricName] = receiver
```

Reply:
Updated 👍